internal/terraform/lang/parser.go

```diff
@@ -115,12 +121,16 @@ func (p *parser) CompletionCandidatesAtPos(file ihcl.TokenizedFile, pos hcl.Pos)
 func (p *parser) BlockTypeCandidates(file ihcl.TokenizedFile, pos hcl.Pos) CompletionCandidates {
 	bTypes := p.blockTypes()
 
-	list := &completeList{
+	list := &candidateList{
 		candidates: make([]CompletionCandidate, 0),
 	}
 
 	prefix := prefixAtPos(file, pos)
 	for name, t := range bTypes {
+		if len(list.candidates) >= p.maxCandidates {
```

Comment:
if the list was exactly max candidates, I think this is a minor bug?

Reply:
I think that's correct, it's just probably slightly more difficult to read/understand, because it's checking the length with the addition from the previous iteration.
If the length is equal to max then this conditional doesn't even get executed, because we reached the end of the slice. I will add one more test to cover this scenario.